controllers/home_controller.rb

```
@@ -53,16 +45,37 @@ def create_catalog
+    
+    def safe_parse(value)
+      return nil unless value
+    
+      parse_item = ->(item) {
+        begin
+          parsed = JSON.parse(
+            item.gsub(/:(\w+)=>/, '"\1":').gsub('=>', ':').gsub('\"', '"')
+          )
+          yield(parsed) if block_given?
+          parsed
+        rescue JSON::ParserError => e
```

RuboCop (1.73)

[warning] Useless assignment to variable - `e`.

(Lint/UselessAssignment)